### ISSUE TITLE
Rework RemoveCounterCost

### ISSUE THREAD

**Issue by Susucre** [magefree/mage#12216]
`RemoveCounterCost` is a cost for removing a number of (specific type) counters from permanents or cards.

Currently, it does take a non-specific `Target`. Notably there is no constraint/check that the target has at least one counter. Which results in an unintuitive UI that rollbacks the payment if selecting one with none of the counter.

It would make a lot more sense to provide Filter(s) instead of Target, and add the extra minimum constraint to the filter to target permanents/cards with at least 1 matching counter there.
It could also make sense to split the class for FilterCard (rare, e.g. [[Mari, the Killing Quill]]) and FilterPermanent (more common, e.g. [[Falco Spara]]).

Another solution (worst in my opinion), would be to always make sure that the target is filtering permanents/cards with at least 1 of the matching counter, but that seems like a lot of extra code in userrs that should be shared by the common effect.

**Comment by JayDi85** [comment]
Yes, it must be improved. Each related cost usage must be carefully reviewed -- if you put filter to current target then it will fail due unsupported `canPay` method (see zero counters use case):

![shot_240502_202553](https://github.com/magefree/mage/assets/8344157/a36a136f-4f89-49c6-a96d-f1385f5b441f)
![shot_240502_202603](https://github.com/magefree/mage/assets/8344157/4f323a03-c2ff-4b28-ba6b-60f11b2801f5)

P.S. Not related to issue -- looks like `MariTheKillingQuill` and `OblivionSower` uses wrongly face-up predicate (I don't see any requirements for it):

![shot_240502_203219](https://github.com/magefree/mage/assets/8344157/4ead5f2c-1591-4fae-8f06-abc793fc6791)

![shot_240502_203030](https://github.com/magefree/mage/assets/8344157/304eaba1-672f-4902-a3d3-3ee1a8da1ed3)
